concourse/model/step.py

```diff
@@ -405,6 +426,35 @@ def retries(self):
     def notification_policy(self):
         return self._notification_policy
 
+    def _on_abort(self):
+        return self.raw.get('on_abort', None)
+
+    def _on_abort_argv(self):
+        on_abort = self._on_abort()
+
+        if on_abort is None:
+            return None
+
+        if not isinstance(on_abort, list):
+            return [str(on_abort)]
+        return [shlex.quote(str(e)) for e in on_abort]
+
+    def on_abort_executable(self, prefix=''):
+        if not self._on_abort_argv():
+            raise RuntimeError("'on_abort' not configured")
+
+        if isinstance(prefix, str):
+            prefix = [prefix]
+        return os.path.join(*prefix, self._on_abort_argv()[0])
+
+    def on_abort_execute(self, prefix=''):
+        if not self._on_abort_argv():
+            raise RuntimeError("'on_abort' not configured")
```

Comment:
if this happened, this would imply there is a bug in our template, correct? if I am correct, a brief stmt about this would be helpful (so we know it's not a cfg error potentially stemming from user-cfg)